### ISSUE TITLE
tt6856242 result year = 0

### ISSUE THREAD

**Issue by allanpk716** [StalkR/imdb#8]
search IMDB ID: tt6856242, result year = 0, wanted year = 2021

**Comment by StalkR** [comment]
Hey, thanks for the report.

Strange, it works from here, e.g. https://movie-db-api.appspot.com/title/tt6856242

I also just pushed a test demonstrating it: 199a89ea305100401d65cbc8a0960cc77d905d7b https://github.com/StalkR/imdb/blob/master/title_test.go#L287

The problem might be region-specific, as IMDb website may respond different pages depending on the language it determines from the querying IP. Feel free to share the page output that you observe from your location and we can see to tweak the parsing to accomodate it.

**Comment by allanpk716** [comment]
When I tested at home last night, there was something wrong when I got Year, and it was normal when I tested in the company just now. I'll try again when I get home and send out the HTML file for the corresponding page.

**Comment by allanpk716** [comment]
My above tests were done using the latest code from your project and are correct. Then I still have problems using V1.0.7 in my project, then I switch your project to V1.0.7 and the problem can recur.

Attached are screenshots and two pages to parse

## screenshots
![image](https://user-images.githubusercontent.com/1176000/164950972-3569c5d9-9683-4caa-b294-9236696d4745.png)

## Attached File
[tt6856242--html file.zip](https://github.com/StalkR/imdb/files/8548834/tt6856242--html.file.zip)

**Comment by allanpk716** [comment]
I see you released V1.0.8, I will upgrade first, in the latest version can be correctly resolved. thank you

**Comment by StalkR** [comment]
Thanks for confirming it works, and also sharing the html.

I verified locally and confirms it works with that html as well:

```
func TestTitleData(t *testing.T) {
	titlePage, err := ioutil.ReadFile("testdata/tt6856242.html")
	if err != nil {
		t.Fatal(err)
	}
	rlsPage, err := ioutil.ReadFile("testdata/tt6856242-releaseinfo.html")
	if err != nil {
		t.Fatal(err)
	}
	got := &Title{}
	if err := got.Parse(titlePage); err != nil {
		t.Fatal(err)
	}
	if err := got.ParseRls(rlsPage); err != nil {
		t.Fatal(err)
	}
	want := Title{
		ID:          "tt6856242",
		URL:         "https://www.imdb.com/title/tt6856242",
		Name:        "The King's Man",
		Type:        "Movie",
		Year:        2021,
		Rating:      "6.3",
		RatingCount: 115757,
		Duration:    "2h11m",
		Directors: []Name{
			Name{ID: "nm0891216", URL: "https://www.imdb.com/name/nm0891216", FullName: "Matthew Vaughn"},
		},
		Writers: []Name{
			Name{ID: "nm0891216", URL: "https://www.imdb.com/name/nm0891216", FullName: "Matthew Vaughn"},
			Name{ID: "nm2244980", URL: "https://www.imdb.com/name/nm2244980", FullName: "Karl Gajdusek"},
			Name{ID: "nm2092839", URL: "https://www.imdb.com/name/nm2092839", FullName: "Mark Millar"},
		},
		Actors: []Name{
			Name{ID: "nm0000146", URL: "https://www.imdb.com/name/nm0000146", FullName: "Ralph Fiennes"},
			Name{ID: "nm2605345", URL: "https://www.imdb.com/name/nm2605345", FullName: "Gemma Arterton"},
			Name{ID: "nm0406975", URL: "https://www.imdb.com/name/nm0406975", FullName: "Rhys Ifans"},
		},
		Genres:        []string{"Action", "Adventure", "Thriller"},
		Languages:     []string{"English", "Latin", "German", "French", "Russian"},
		Nationalities: []string{"United Kingdom", "United States"},
		Description:   "In the early years of the 20th century, the Kingsman agency is formed to stand against a cabal plotting a war to wipe out millions.",
		Poster:        Media{ID: "rm1627514625", TitleID: "tt6856242", URL: "https://www.imdb.com/title/tt6856242/mediaviewer/rm1627514625", ContentURL: "https://m.media-amazon.com/images/M/MV5BMjg2YTdhZGQtZTA4ZS00MTVmLTg1NWUtZWNkNzE5M2IzNDkzXkEyXkFqcGdeQXVyNzI1NzMxNzM@._V1_.jpg"},
		AKA:           []string{"King's Man. Pradžia", "King's Man: A Origem", "King's Man: A kezdetek", "King's Man: El origen", "King's Man: Pierwsza misja", "King's Man: Начало", "King's Man: Първа мисия", "Kings man: Ha'Hatkhala", "Kings's Man: Početak", "Kingsman 3", "Kingsman: First Agent", "Kingsman: Khởi Nguồn", "Kingsman: O Grande Jogo", "Kingsman: Première mission", "Kingsman: První mise", "Kingsman: The Great Game", "The Beginning", "The King's Man", "The King's Man - Le origini", "The King's Man - The Beginning", "The King's Man: Başlangıç", "The King's Man: Khởi Nguồn", "The King's Man: La primera misión", "The King's Man: O Início", "The King's Man: Première mission", "The King's Man: Prvá misia", "The King's Man: Zacetek", "The King's Man: Începutul", "The King's Man: Почетак", "The king's man: pirmā misija", "Ο Άνθρωπος του Βασιλιά: Το Ξεκίνημα", "Кінгс Мен", "Кінґс Мен", "キングスマン：ファースト・エージェント", "皇家特工：第一任務", "金牌特務：金士曼起源", "킹스맨: 퍼스트 에이전트"},
	}
	if got.RatingCount > want.RatingCount && want.RatingCount > 0 {
		want.RatingCount = got.RatingCount
	}
	if err := diffStruct(*got, want); err != nil {
		t.Errorf("Title(%s): %v", want.ID, err)
	}
}
```

```
$ go clean -testcache; go test -v . -run 'TestTitleData'
=== RUN   TestTitleData
--- PASS: TestTitleData (0.08s)
PASS
ok      github.com/StalkR/imdb  0.606s
```
